app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,83 @@
+class VendorsController < ApplicationController
+  def index
+    uri = URI(rmapi_path)
+
+    # Transform query params to what the EBSCO RM-API expects
+    uri.query = URI.encode_www_form(
+      search: params[:q],
+      orderby: params[:orderby] || params[:q] ? 'relevance' : 'vendorname',
+      count: params[:count] || 25,
+      offset: params[:offset] || 1
+    )
+
+    # Make the RM-API request and parse the response body
+    response = make_request(uri)
+    data = JSON.parse(response.body)
+
+    # Return the list of vendors
+    if response.code === '200'
+      vendors = data['vendors'].map do |vendor|
+        Vendor.new(vendor)
+      end
+
+      render jsonapi: vendors,
+             meta: { totalResults: data['totalResults'] },
+             status: :ok
+    else
+      # Map RM-API errors to JSON-API error objects
+      errors = data['Errors'].map { |err| { title: err['Message'] } }
```

Comment:
`Map` or `Hashugar` could add a little syntactic sugar to your `data` hash.  Even without bringing those in, i think `JSON.parse(...)` accepts some kind of flag like `symbolize_keys: true` which could at least free you up from `data['someString']['someOtherString']` and set you up with some symbols for keys.  All of that is relatively minor, but as we start tearing through more of these endpoints it's worth keeping in mind.